### DISCUSSION TITLE
Long way to "go" and cronjobs don't work

### DISCUSSION THREAD

**Discussion by SchinkTasia** [firefly-iii/firefly-iii#5840] [Firefly III]
Hello,
I searched for a finance web based software and here I am :)

I have several troubles installing Firefly, so it took me yesterday 4 to 5 hours until I could access the website.
(no, it wasn't a problem with my apache2, the docker and docker-compose images don't worked for me.) My solution was to use the Alpha and not the latest. The alpha starts and installs without errors.

My actual problem is, that I installed the docker cron job (vie crontab and call the docker-compose container solution the wiki offer) but actually firefly tell me, that the cronjobs never runs and that this is normal for a first time installation. And yes I startet the cronjobs but don't work.

Any suggestions?

Best regards

**Comment by JC5** [comment]
I can't help you if you're running an alpha version. The latest alpha came out ages ago, which means that particular version is ancient. 

I'd much rather find out why the latest release doesn't work for you.

**Reply by SchinkTasia** [reply]
Hello,
That's totally fine.

At the beginning I got the error "Acces denied for user".
The app couldn't login via mysql. So start building around and edit the user in mysql so, that the could verify.

At this point I got many errors, because the latest build could not found tables at startup. I mean yes, there are no tables because I started the installation of firefly. But firefly recognized that there are no tables but don't start to create these.
At the end the firefly app said "Thank you for installation" but it apperrntlx don't work, because of all missing tables In the database.

And yes, the user had fullrights for the database.

I hope that helps you.

**Reply by JC5** [reply]
Not really. Firefly III is capable of creating and managing the tables. Even when it previously failed. I’m not really sure what went wrong unfortunately. 

Either way, if you start over and delete the volumes as well it should run as expected. That’s the advantage of Docker

**Reply by SchinkTasia** [reply]
Yes, I know it should. But after the 4th time deleting and installing again and running in the same error, I stopped trying it :(

I can try it again and send you here the logs, if you want.
But I can do this first in like 11h.

**Reply by JC5** [reply]
Sure thing, please do.

**Reply by SchinkTasia** [reply]
Hello,

This is the error i got after i start the docker-compose installation.

```
Whoops! An error occurred.

Unfortunately, this error was not recoverable :(. Firefly III broke. The error is:

Could not poll the database: SQLSTATE[42S02]: Base table or view not found: 1146 Table 'firefly.configuration' doesn't exist (SQL: select `id`, `name`, `data` from `configuration` where `name` = is_demo_site and `configuration`.`deleted_at` is null limit 1)
More information

Please collect more information in the storage/logs directory where you will find log files. If you're running Docker, use docker logs -f [container]. You can read more about collecting error information in [the FAQ](https://docs.firefly-iii.org/faq/other#how-do-i-enable-debug-mode).
Get help [on GitHub](https://github.com/firefly-iii/firefly-iii/issues)

You're more than welcome to open a new issue on GitHub.

   1. Use the search!
   2. Include the information [from this debug page](https://firefly.schinktasia.de/debug).
   3. Tell us more than "it says Whoops!"
   4. Include error logs (see above).
   5. Tell us what you were doing.
```

**Reply by SchinkTasia** [reply]
So,
Explizit this is standing in the console:

`Could not poll the database: SQLSTATE[42S02]: Base table or view not found: 1146 Table 'firefly.configuration' doesn't exist (SQL: select `id`, `name`, `data` from `configuration` where `name` = is_demo_site and `configuration`.`deleted_at` is null limit 1) `

Mutltiple times with other tables.

**Reply by JC5** [reply]
Could you share the logs from the start of the container onwards?

**Reply by SchinkTasia** [reply]
Here:

`Now in entrypoint.sh for Firefly III
Script:       1.0.21 (2022-02-17)
User:         'root'
Group:        'root'
Working dir:  '/var/www/html'
Build number:  106
Build date:    28-02-2022 08:34:56 CET
Now parsing _FILE variables.
done!
Touch DB file (if SQLite)...
Will not download LDAP packages.
Dump auto load...
Generating optimized autoload files
Generated optimized autoload files containing 5384 classes
Discover packages...
Discovered Package: diglactic/laravel-breadcrumbs
Discovered Package: fideloper/proxy
Discovered Package: jc5/google2fa-laravel
Discovered Package: laravel/passport
Discovered Package: laravel/sanctum
Discovered Package: laravel/ui
Discovered Package: laravelcollective/html
Discovered Package: nesbot/carbon
Discovered Package: rcrowe/twigbridge
Package manifest generated successfully.
Current working dir is '/var/www/html'
Wait for the database.
DB is up. Time to execute artisan commands.
Wait another 5 seconds in case the DB needs to boot.
Done waiting for the DB to boot.
Current working dir is '/var/www/html'
Run various artisan commands...
Running migration commands...
DSN is mysql:host=fireflyiiidb;port=3306;charset=utf8mb4
Database "firefly" exists.
Migration table created successfully.
[2022-03-02 12:40:07] local.ERROR: Exception is: {"class":"Symfony\\Component\\ErrorHandler\\Error\\FatalError","errorMessage":"Cannot declare class CreatePersonalAccessTokensTable, because the name is already in use","time":"Wed, 02 Mar 2022 12:40:07 +0100","file":"\/var\/www\/html\/database\/migrations\/2021_12_27_000001_create_personal_access_tokens_table.php","line":11,"code":0,"version":"5.6.15","url":"https:\/\/firefly.schinktasia.de","userAgent":"Symfony","json":true}  
[2022-03-02 12:40:07] local.ERROR: Cannot declare class CreatePersonalAccessTokensTable, because the name is already in use {"exception":"[object] (Symfony\\Component\\ErrorHandler\\Error\\FatalError(code: 0): Cannot declare class CreatePersonalAccessTokensTable, because the name is already in use at /var/www/html/database/migrations/2021_12_27_000001_create_personal_access_tokens_table.php:11)
[stacktrace]
#0 {main}
"} 

In 2021_12_27_000001_create_personal_access_tokens_table.php line 11:
                                                                               
  Cannot declare class CreatePersonalAccessTokensTable, because the name is a  
  lready in use                                                                
                                                                               

Going to decrypt the database.`

**Reply by SchinkTasia** [reply]
When you need it, here is the whole starting log until `Thank you for installing Firefly III, v5.6.15!`
[Log.txt](https://github.com/firefly-iii/firefly-iii/files/8169428/Log.txt)

**Reply by JC5** [reply]
Thanks. If you upgrade to .16 you should no longer see this error.

**Reply by SchinkTasia** [reply]
I will try this. Thank you 👍

**Reply by SchinkTasia** [reply]
How do i install it via docker? with the latest tag?

**Reply by JC5** [reply]
Yes.

**Reply by SchinkTasia** [reply]
Installation succesfull ;)
